### ISSUE TITLE
git2net throws AssertionError when mining a repository

### ISSUE THREAD

**Issue by DiegoEliasCosta** [gotec/git2net#8]
**Bug description**
Cannot finish mining the repository [gotk3/gotk3] (https://github.com/gotk3/gotk3) as the tool crashes with an AssertionError on the commit parents not been equal to 2.  

**To Reproduce**
1. Clone the gotk3/gotk3 repository (Golang project)
2. Attempt to mine the repository 
3. See error
```
"""
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/git2net/extraction.py", line 1073, in _process_commit
    df_edits = df_edits.append(_extract_edits_merge(git_repo, commit,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/git2net/extraction.py", line 693, in _extract_edits_merge
    assert len(commit.parents) == 2
AssertionError
"""
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/git2net/extraction.py", line 1539, in mine_git_repo
    _process_repo_parallel(repo_string=repo_string, sqlite_db_file=sqlite_db_file,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/git2net/extraction.py", line 1220, in _process_repo_parallel
    for result in p.imap_unordered(_process_commit, args, chunksize=chunksize):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py", line 865, in next
    raise value
AssertionError
```
4. After running the tool, it seems git2net is leaking some semaphores. They need to be closed even in case the assertion fails.
```
/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 8 leaked semaphore objects to clean up at shutdown
``` 

**Desktop (please complete the following information):**
 - OS: MacOS Catalina (10.15.3)
 - Python 3.8.1
 - git2net version 1.1.7

**Additional context**
This might be due to a particularity on the git branch structure of the project, which I wouldn't know anything about. This was, just by coincidence, the first project I tried to run the tool with.

**Comment by gotec** [comment]
Hi Diego
Thank you for letting me know. So far I‘ve not come across a commit with more than 2 parents. This is quite peculiar and currently not supported by git2net. I didn’t think they could exist but glad I wrote the assertion.
I will look into this next week.
Cheers,
Christoph

**Comment by gotec** [comment]
The issues with the semaphores should have originated from the error occurring inside the multiprocessing pool. In this case the the run is terminated before the pool is properly closed. If this issue persists after a successful run (either after I fix the issue above or for another repository) please let me know.

**Comment by DiegoEliasCosta** [comment]
Just a suggestion, wouldn't it be better to catch all runtime issues and make sure to close the pool? There are no lasting effects, AFAIK, but would prevent this resource from leaking and eventually new issues been opened for you to fix here ;)

The tool is very cool, btw! Will try to use it in the near future :)

**Comment by gotec** [comment]
I have updated the code on github to account for merges with more than 2 parents. This resolves the issue and a crawl for gotk3 runs without any errors. Over the next days I will test that merges are still processed properly before I make a new release of pypi.

**Comment by DiegoEliasCosta** [comment]
Thanks @gotec! Let me know when it makes the release of PyPI so I can also test with another bunch of repositories. 

I will close it for now.

**Comment by gotec** [comment]
Fixed in git2net 1.1.8 on PyPI

**Comment by gotec** [comment]
I have just tested the released version and can confirm no errors on 'gotk3'. According to my test setup also the results are as expected.

Let me know in case you still have trouble.

Cheers!
